base/docs/helpdb/Base.jl

```diff
@@ -8908,19 +8856,17 @@ vecnorm
 """
     isvalid(value) -> Bool
 
-Returns `true` if the given value is valid for its type, which currently can be one of
-`Char`, `String`, `UTF16String`, or `UTF32String`.
+Returns `true` if the given value is valid for its type, which currently can be either
+`Char` or `String`.
```

Comment:
If IJulia needs UTF-8 validation, it may need to use LegacyStrings. Otherwise we could have an `isvalid(String)` method that returns true if the `String` is valid UTF-8 and false otherwise. The tricky thing is that there are a few different versions of what could be considered valid:

- encoding sanity (leading byte followed by the corresponding number of trailing bytes)
- invalid code points (surrogates)
- overlong encodings

And of course, these are not mutually exclusive – each string can exhibit any subset of these issues.